### PR TITLE
refac(config): add AutoDetectErrors and AutoTrackSessions and soft deprecate predecessors

### PR DESCRIPTION
**Pull request by rich-bugsnag**
## Goal

add AutoDetectErrors and AutoTrackSessions and soft deprecate predecessors

## Changeset

Changed names in relevant places and added deprication warnings to old fields.

Also used the FormerlySerializedAsAttribute in the bugsnage behaviour script so that people updating wont loose their settings on the gameobject

## Testing

Manually tested and updated unit tests

### CODE REVIEW

File: src/BugsnagUnity/Bugsnag.cs [hunk 1]
```diff
@@ -91,12 +92,24 @@ public static void SetContext(string context)
         /// <summary>
         /// By default, we will automatically notify Bugsnag of any fatal errors (crashes) in your game.
         /// If you want to stop this from happening, you can set the AutoNotify property to false. It
-        /// is recommended that you set this value by Configuration rather than this method.
+        /// is recommended that you set this value by Configuration at init rather than this method.
         /// </summary>
         /// <param name="autoNotify"></param>
+        [Obsolete("SetAutoNotify is depricated, please use SetAutoDetectErrors instead.", false)]
```
[the remark this answers is not in the document]
Comment (kattrali) [reply src/BugsnagUnity/Bugsnag.cs:98]: As an aside, do these annotations show up in an editor (as a build warning) or do the interface functions need to be marked as well?

**Review by kattrali — COMMENTED** [review]
looks good overall, added a few comments

File: src/BugsnagUnity/Bugsnag.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (rich-bugsnag) [reply src/BugsnagUnity/Bugsnag.cs]: Adding the warning here works well 👍

File: src/BugsnagUnity/Client.cs [hunk 2]
```diff
@@ -382,13 +382,13 @@ public void SetContext(string context)
             NativeClient.SetContext(context);
         }
 
-        public void SetAutoNotify(bool autoNotify)
+        public void SetAutoDetectErrors(bool autoDetectErrors)
         {
             // set the AutoNotify property on Configuration, as it currently controls whether C# errors are reported
```
[the remark this answers is not in the document]
Comment (rich-bugsnag) [reply src/BugsnagUnity/Client.cs]: done

**Review by kattrali — APPROVED** [review]

**Pull request merged**